Wikipedia/UI-V5/WMFArticleListCollectionViewController.m

```diff
@@ -198,43 +219,54 @@ - (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id <UIV
     } completion:NULL];
 }
 
-#pragma mark - UICollectionViewDelegate
+#pragma mark - Article Selection
 
-- (void)collectionView:(UICollectionView*)collectionView didSelectItemAtIndexPath:(NSIndexPath*)indexPath {
-    self.selectedArticle = [self.dataSource articleForIndexPath:indexPath];
+- (void)showArticle:(MWKArticle*)article discoveryMethod:(MWKHistoryDiscoveryMethod)discoveryMethod {
+    if (self.delegate) {
+        [self.delegate didSelectArticle:article sender:self];
+        return;
+    }
 
     WMFArticleContainerViewController* container =
```

Comment:
Is the below logic everywhere it needs to be (Mostly referring to the page logging)? 
Should we move article presentation into a category as well so it can be shared among all VCs? That way we don't have divergent presentation logic?